src/common/transformations/src/transformations/transpose_sinking/ts_gather.cpp

```diff
@@ -260,7 +260,11 @@ TSGatherBackward::TSGatherBackward() {
                 main_node->input(1).replace_source_output(squeeze->input_value(0));
             }
         }
+        std::vector<size_t> new_axes_val;
         if (!axes_val.empty()) {
+            for (size_t i = 0; i < axes_val.size(); ++i) {
+                new_axes_val.push_back(order_val[axes_val[i]]);
```

Comment:
can we expect out-of-range error here?

Reply:
I don't think so. If you take a look at how axes_val is constructed - its values can be from following range [0, shape.size()]. `shape` is also an input shape to Transpose, so order_val.size() == shape.size().